grails-app/services/com/recomdata/transmart/data/export/ClinicalExportService.groovy

```diff
@@ -85,9 +106,40 @@ class ClinicalExportService {
                 csvWriter.writeNext(rowList as String[])
             }
         }
+
         clinicalDataFile
     }
 
+    File exportAllTags(Set<OntologyTerm> terms, File studyDir) {
+        def tagsMap = ontologyTermTagsResourceService.getTags(terms, true)
+
+        if (tagsMap) {
+            def resultFile = new File(studyDir, META_FILE_NAME)
+
+            resultFile.withWriter { Writer writer ->
+                CSVWriter csvWriter = new CSVWriter(writer, COLUMN_SEPARATOR)
+                csvWriter.writeNext(META_FILE_HEADER as String[])
+                tagsMap.each { OntologyTerm keyTerm, List<OntologyTermTag> valueTags ->
+                    valueTags.each { OntologyTermTag tag ->
+                        csvWriter.writeNext([keyTerm.fullName, tag.name, tag.description] as String[])
+                    }
+                }
+            }
+
+            resultFile
+        }
+    }
+
+    private Set<OntologyTerm> getOntologyTerms(List<NormalizedLeafsVariable> variables) {
+        variables.collectMany { NormalizedLeafsVariable variable ->
+            variable.innerClinicalVariables.collect {
+                def conceptFullName = new ConceptFullName(it.conceptPath)
+                def conceptKey = new ConceptKey(conceptFullName.parts[0], conceptFullName.toString())
+                conceptsResourceService.getByKey(conceptKey.toString())
+            }
+        } as Set
```

Comment:
Could make more sense to add `ClinicalVariable::getOntologyTerm()` and then return `composedVariables.collectMany { it.innerClinicalVariables }*.ontologyTerm as Set` here. I also think this logic won't work with across trial nodes.